pkg/server/server.go

```diff
@@ -44,18 +48,46 @@ func New(conf *config.Configurations) (*BCDBHTTPServer, error) {
 		return nil, err
 	}
 
-	db, err := bcdb.NewDB(conf, lg)
+	var metricsRegistry *prometheus.Registry
+	var metricsServer *http.Server
+	var metricsListen net.Listener
+	if conf.LocalConfig.Prometheus.Enabled {
+		metricsRegistry = prometheus.NewRegistry()
+		metricsRegistry.MustRegister(
+			prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}),
+			prometheus.NewGoCollector(),
+		)
```

Comment:
how do units that get the registry behave if it is nil? do they still collect metrics? i.e. counters, measure time, etc?

Reply:
When the registry is nil, the metrics object methods do nothing.